### PR TITLE
Checkpoint on SIGTERM

### PR DESCRIPTION
**Pull request by roelof-groenewald**
Seems to work pretty well so far.

Still need to update the changelog and version number.

### CODE REVIEW

**Review by peterscherpelz — COMMENTED** [review]
Would it be worth adding an 'installaftercheckpoint' python hook so that we can just always run our python stuff after any warpx checkpoint, rather than having to reconfirm steps in our function too?

And have you checked that updating the fullhist_dict at non-standard timepoints and then restarting works correctly? I don't know that it doesn't, I'm just suspcious :)

**Comment by roelof-groenewald** [comment]
@peterscherpelz Yeah probably is worth adding such a callback but I'm not sure it is worth waiting for it before we pull this functionality into our AWS simulations.

I did check that after a restart from an intermediate checkpoint the consequent flux diagnostics were still as expected but I'm not sure I considered all possible scenarios. If you have specific testing ideas in mind I'd be happy to do those.

File: mewarpx/changelog.csv [hunk 1]
```diff
@@ -1,4 +1,15 @@
 Version, Physics version, Date,        List of changes
+7.2.0, 2, In progress, "
+
+**API Changes**:
+
+- :class:`mewarpx.sim_control.SimControl` now takes optional arguments
+  ``diag_steps``, ``checkpoint`` and ``dump_period`` at initialization. By
+  letting ``SimControl`` install checkpointing, flux daignostics will also be
```
Comment (PhilMiller) [inline mewarpx/changelog.csv:8]:
```suggestion
  letting ``SimControl`` install checkpointing, flux diagnostics will also be
```

**Comment by roelof-groenewald** [comment]
Closing this due to better signal handling implemented by Phil.

**Pull request closed without merging**